pkg/deploy/lattice/access_log_subscription_manager.go

```diff
@@ -72,58 +73,101 @@ func (m *defaultAccessLogSubscriptionManager) Create(
+		 * If it is the same ALP, return success. Else, return ConflictError.
+		 */
+		listALSInput := &vpclattice.ListAccessLogSubscriptionsInput{
+			ResourceIdentifier: &resourceIdentifier,
+		}
+		listALSOutput, err := vpcLatticeSess.ListAccessLogSubscriptionsWithContext(ctx, listALSInput)
+		if err != nil {
+			return nil, err
+		}
+		for _, als := range listALSOutput.Items {
+			if *als.DestinationArn == accessLogSubscription.Spec.DestinationArn {
+				listTagsInput := &vpclattice.ListTagsForResourceInput{
+					ResourceArn: als.Arn,
+				}
+				listTagsOutput, err := vpcLatticeSess.ListTagsForResourceWithContext(ctx, listTagsInput)
+				if err != nil {
+					return nil, err
+				}
+				value, exists := listTagsOutput.Tags[lattice.AccessLogPolicyTagKey]
+				if exists && *value == accessLogSubscription.Spec.ALPNamespacedName.String() {
+					return &lattice.AccessLogSubscriptionStatus{
+						Arn: *als.Arn,
+					}, nil
 				}
 			}
-			return nil, services.NewConflictError(
-				string(accessLogSubscription.Spec.SourceType),
-				accessLogSubscription.Spec.SourceName,
-				e.Message(),
-			)
-		default:
-			return nil, err
 		}
+		return nil, services.NewConflictError(
+			string(accessLogSubscription.Spec.SourceType),
+			accessLogSubscription.Spec.SourceName,
+			e.Message(),
+		)
+	default:
+		return nil, err
 	}
+}
 
-	return &lattice.AccessLogSubscriptionStatus{
-		Arn: *createALSOutput.Arn,
-	}, nil
+func (m *defaultAccessLogSubscriptionManager) Update(
+	ctx context.Context,
+	accessLogSubscription *lattice.AccessLogSubscription,
+) (*lattice.AccessLogSubscriptionStatus, error) {
+	vpcLatticeSess := m.cloud.Lattice()
+	updateALSInput := &vpclattice.UpdateAccessLogSubscriptionInput{
+		AccessLogSubscriptionIdentifier: aws.String(accessLogSubscription.Status.Arn),
+		DestinationArn:                  aws.String(accessLogSubscription.Spec.DestinationArn),
+	}
+	updateALSOutput, err := vpcLatticeSess.UpdateAccessLogSubscriptionWithContext(ctx, updateALSInput)
+	if err == nil {
+		return &lattice.AccessLogSubscriptionStatus{
+			Arn: *updateALSOutput.Arn,
+		}, nil
+	}
+
+	switch e := err.(type) {
+	case *vpclattice.AccessDeniedException:
+		return nil, services.NewInvalidError(e.Message())
+	case *vpclattice.ResourceNotFoundException:
+		if *e.ResourceType == "SERVICE_NETWORK" || *e.ResourceType == "SERVICE" {
+			return nil, services.NewNotFoundError(string(accessLogSubscription.Spec.SourceType), accessLogSubscription.Spec.SourceName)
+		}
+		return nil, services.NewInvalidError(e.Message())
+	case *vpclattice.ConflictException:
+		/*
+		 * A conflict can happen when the destination type of the new ALS is different from the original.
+		 * To gracefully handle this, we create a new ALS with the new destination, then delete the old one.
+		 */
+		alsStatus, err := m.Create(ctx, accessLogSubscription)
+		if err != nil {
+			return nil, err
+		}
+		err = m.Delete(ctx, accessLogSubscription)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Comment and code contradicts to each other. From code it looks like we create and delete same accessLogSubscription, not "new" and "old" ones.

Reply:
Create uses the ALS Spec to call VPC Lattice Create API

Delete uses the ARN of the ALS to call VPC Lattice Delete API

I could change Delete function to simply accept an ARN, I think that would make more sense anyways